typescript/t.d.ts

```diff
@@ -126,15 +126,15 @@ type ParseKeysByFallbackNs<Keys extends $Dictionary> = _FallbackNamespace extend
     ? Keys[UnionFallbackNs]
     : Keys[_FallbackNamespace & string];
 
-export type FilterKeysByContext<Keys, Context> = Context extends string
+type FilterKeysByContext<Keys, Context> = Context extends string
   ? Keys extends
       | `${infer Prefix}${_ContextSeparator}${Context}${_PluralSeparator}${PluralSuffix}`
       | `${infer Prefix}${_ContextSeparator}${Context}`
     ? Prefix
     : never
   : Keys;
 
-export type ParseKeys<
```

Comment:
Beware about removing these exports some plugins / users might be using it.
`ParseKeys` is extremely useful also in user land

Reply:
Ah, this was my IDE I think.

This is a quirk of `*.d.ts` files -- they behave like ambient namespaces (other way around actually, ambient namespaces behave like declaration files) in that all types that they contain are implicitly exported.

I'm happy to revert these lines, but just know that all types in `t.d.ts` are available to be used in userland.

Reply:
> I'm happy to revert these lines, but just know that all types in t.d.ts are available to be used in userland.

Yes I know... they were always exported to userland. 
At some point they were moved into separate files to support different version of typescript.
Check the following diff https://github.com/i18next/i18next/pull/2007/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8